src/modules/75-cf/pages/target-detail/components/FlagSettings/FlagSettings.tsx

```diff
@@ -0,0 +1,108 @@
+import { useParams } from 'react-router-dom'
+import { getErrorInfoFromErrorObject, useToaster } from '@harness/uicore'
+import { useStrings } from 'framework/strings'
+import { Feature, Target, useGetAllFeatures, usePatchTarget } from 'services/cf'
+import { ContainerSpinner } from '@common/components/ContainerSpinner/ContainerSpinner'
+import { Page } from '@common/exports'
+import { getErrorMessage } from '@cf/utils/CFUtils'
+import type { TargetManagementFlagConfigurationPanelFormValues as FormValues } from '@cf/components/TargetManagementFlagConfigurationPanel/types'
+import TargetManagementFlagConfigurationPanel from '@cf/components/TargetManagementFlagConfigurationPanel/TargetManagementFlagConfigurationPanel'
+import buildInstructions from './buildInstructions'
+
+export interface FlagSettingsProps {
+  target: Target
+}
+
+const FlagSettings: FC<FlagSettingsProps> = ({ target }) => {
+  const { getString } = useStrings()
+  const { showError } = useToaster()
+  const { accountId: accountIdentifier, projectIdentifier, orgIdentifier } = useParams<Record<string, string>>()
+
+  const {
+    data: flags,
+    loading: loadingFlags,
+    error: flagsError,
+    refetch: refetchFlags
+  } = useGetAllFeatures({
+    queryParams: {
+      accountIdentifier,
+      projectIdentifier,
+      orgIdentifier,
+      environmentIdentifier: target.environment,
+      targetIdentifierFilter: target.identifier,
+      pageSize: 10000 // bring back everything
+    }
+  })
+
+  const initialValues = useMemo<FormValues>(
+    () => ({
+      flags: (flags?.features || []).reduce(
+        (values, { identifier: flagIdentifier, envProperties }) => ({
+          ...values,
+          [flagIdentifier]: {
+            variation: /* istanbul ignore else */ envProperties?.variationMap?.find(({ targets }) =>
+              targets?.find(({ identifier }) => identifier === target.identifier)
+            )?.variation
+          }
+        }),
+        {}
+      )
+    }),
+    [flags?.features, target.identifier]
+  )
+
+  const { mutate: patchTarget } = usePatchTarget({
+    identifier: target.identifier,
+    queryParams: {
+      orgIdentifier,
+      projectIdentifier,
+      accountIdentifier,
+      environmentIdentifier: target.environment
+    }
+  })
+
+  const onChange = useCallback(
+    async (values: FormValues) => {
+      const instructions = buildInstructions(values, initialValues)
+
+      /* istanbul ignore else */
+      if (instructions.length) {
+        try {
+          await patchTarget({ instructions })
+          refetchFlags()
+        } catch (e) {
+          showError(getErrorInfoFromErrorObject(e))
+        }
+      }
+    },
+    [initialValues, patchTarget, refetchFlags, showError]
+  )
+
+  if (loadingFlags) {
+    return <ContainerSpinner flex={{ align: 'center-center' }} />
+  }
+
+  if (flagsError) {
+    return <Page.Error message={getErrorMessage(flagsError)} onClick={() => refetchFlags()} />
```

Comment:
In the onClick, could you call "refetchFlags" directly without the anonymous function?

```suggestion
    return <Page.Error message={getErrorMessage(flagsError)} onClick={refetchFlags} />
```

Reply:
Looking at the `PageError` component, the `onClick` is just passed on directly from a `Button` component, so it'll pass the click event as the 1st argument. The `refetchFlags` function accepts an argument to override the original request params. So doing it direct would pass the click event as the override request params and who knows what it would do then 🤪

Reply:
Only one way to find out 💥